diff -urN busybox-1.33.1.orig/loginutils/login.c busybox-1.33.1/loginutils/login.c
--- busybox-1.33.1.orig/loginutils/login.c	2021-01-01 02:52:27.000000000 -0800
+++ busybox-1.33.1/loginutils/login.c	2021-11-30 09:30:33.436537560 -0800
@@ -54,12 +54,13 @@
 //kbuild:lib-$(CONFIG_LOGIN) += login.o
 
 //usage:#define login_trivial_usage
-//usage:       "[-p] [-h HOST] [[-f] USER]"
+//usage:       "[-p] [-h HOST] [[-f] USER] [-n]"
 //usage:#define login_full_usage "\n\n"
 //usage:       "Begin a new session on the system\n"
 //usage:     "\n	-f	Don't authenticate (user already authenticated)"
 //usage:     "\n	-h HOST	Host user came from (for network logins)"
 //usage:     "\n	-p	Preserve environment"
+//usage:     "\n	-n  Allow password less root logins"
 
 #include "libbb.h"
 #include "common_bufsiz.h"
@@ -317,6 +318,7 @@
 		LOGIN_OPT_f = (1<<0),
 		LOGIN_OPT_h = (1<<1),
 		LOGIN_OPT_p = (1<<2),
+		LOGIN_OPT_n = (1<<3),
 	};
 	char *fromhost;
 	char username[USERNAME_SIZE];
@@ -356,7 +358,7 @@
 	bb_daemon_helper(DAEMON_CLOSE_EXTRA_FDS);
 
 	username[0] = '\0';
-	opt = getopt32(argv, "f:h:p", &opt_user, &opt_host);
+	opt = getopt32(argv, "f:h:pn", &opt_user, &opt_host);
 	if (opt & LOGIN_OPT_f) {
 		if (!run_by_root)
 			bb_simple_error_msg_and_die("-f is for root only");
@@ -403,6 +405,9 @@
 			get_username_or_die(username, sizeof(username));
 
 #if ENABLE_PAM
+		if (!strncmp(username, "root", 4) && (opt & LOGIN_OPT_n))
+			opt |= LOGIN_OPT_f;
+
 		pamret = pam_start("login", username, &conv, &pamh);
 		if (pamret != PAM_SUCCESS) {
 			failed_msg = "start";
@@ -486,6 +491,10 @@
 		if (pw->pw_passwd[0] == '!' || pw->pw_passwd[0] == '*')
 			goto auth_failed;
 
+		/* Force password less root authentication if specifically requested */
+		if (pw->pw_uid == 0 && (opt & LOGIN_OPT_n))
+			opt |= LOGIN_OPT_f;
+
 		if (opt & LOGIN_OPT_f)
 			break; /* -f USER: success without asking passwd */
 
